public/js/logic.js

```diff
@@ -31,14 +34,27 @@ $("#container").on("click", "div.chest-wrapper", function(e){
   $(this).hide("fast", done(e));
 });
 
-function addChests(count) {
+function addChests(count, timer) {
+  let timeOut = timer * 60 * 1000 // Convert input in minutes to ms.
   for(let i = 1; i<=count; i++){
     container.append(chest);
     chestCount++;
+
+    if (timer && timer != 0){
+      let lastChest = container.children().last();
+      setTimeout(function() {deleteSpecificChest(lastChest);}, timeOut);
```

Comment:
Need to clear/nuke any ongoing timers if those chests are removed prematurely via the clear or remove actions:

1. Add 10 chests.
2. Add 2 chests with a 1 minute timer.
3. Remove 6 chests manually (click on them or use remove button) immediately
4. After step (2) above timer is finished, two MORE chests disappear (unexpected/unintentional)

Maybe we need unique ids on these batches when they are added?

Reply:
Uniqueness (id with unix timestamp+counter) added and tested.
1. When a specific chest is deleted, the timeout is cleared
2. When timeOut ends, check if its still in the DoM, only then run the remove function  (should not even run for removed chests)
3. After a 'Clear' all chests, we cant globally clear timeouts. so 2. kicks in as failsafe